idaes/core/solvers/petsc.py

```diff
@@ -544,6 +544,7 @@ def calculate_time_derivatives(m, time):
                 for i, v in var.items():
                     try:
                         if disc_eq[i].active:
+                            v.value = 0 # Make sure there is a value
```

Comment:
Would it be worth doing:

```
if v.value is None:
    v.value = 0
```

As it stands, this will overwrite any value that is already there, even if it is already a good guess.

Reply:
Yeah, I wasn't sure how to handle this.  I don't really know what's good and what isn't.  The calculation is pretty straight forward, but somehow the derivative vars have random junk that is maybe NaN.  I think it's a side affect of not returning derivative values from PETSc, but having derivative variables.

I guess the best solution to this is to make PETSc estimate and return derivatives then use this routing to adjust them given that the time steps in the descretized problem are different.  I guess this should be okay until I can update the solver.